packages/message-relayer/src/service.ts

```diff
@@ -238,16 +251,26 @@ export class MessageRelayerService extends BaseService<MessageRelayerOptions> {
           }
 
           const receipt = await this._relayMultiMessageToL1(
-            this.state.messageBuffer
+            this.state.messageBuffer.reduce((acc, cur) => {
+              acc.push(cur.payload)
+              return acc
+            }, [])
           )
 
           console.log('Receipt:', receipt)
 
           /* parse this to make sure that the mesaage was actually relayed */
-          if (/*everything went well*/ true) {
+          // clear out buffer only if the messages are relayed to L1 successfully
+          if (
+            await this._wasMessageRelayed(this.state.messageBuffer[0].message)
+          ) {
```

Comment:
there could be an interesting scenario here, wasMessageRelayed() would return true for only successful confirmed relays, over just relay tx attempt (the status of which then depends on the method/target that is being called, so could be made to fail purposely). now if the array is not made empty, i.e. we don't skip for such, continuous failed attempts will be made until stopped.

Reply:
We could have this issue when we restart the message relayers, too.